src/components/CheckableInput/CheckableInput.tsx

```diff
@@ -0,0 +1,52 @@
+import classNames from "classnames";
+import { TSFixMe } from "index";
+import { nanoid } from "nanoid";
+import PropTypes from "prop-types";
+import React, { useEffect, useRef, HTMLProps } from "react";
+
+type Props = {
+  type: string;
+  label: string;
+  required?: boolean;
+  indeterminate?: boolean;
+} & HTMLProps<HTMLElement>;
+
+const CheckableInput = ({
+  type,
+  label,
+  required = false,
+  indeterminate = false,
+  ...checkboxProps
+}: Props): JSX.Element => {
+  const inputId = useRef(nanoid());
+  // setting this type as any for now, otherwise it causes linting errors
+  const inputRef: TSFixMe = useRef(null);
+
+  useEffect(() => {
+    inputRef.current.indeterminate = indeterminate;
+  }, [indeterminate]);
+
+  return (
+    <label className={classNames(`p-${type}`, { "is-required": required })}>
+      <input
+        ref={inputRef}
+        type={type}
+        aria-labelledby={inputId.current}
+        className={classNames(`p-${type}__input`)}
```

Comment:
you're not really using `classNames` here :) 
```suggestion
        className={`p-${type}__input`}
```

Reply:
Thanks for the review, and all the pointers - this is one of my first encounters with React and TypeScript, so it's very useful. I'm gonna resolve some of these comments without comment while implementing the suggestions - just imagine me face palming and saying "of course..." :laughing: